src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,296 @@
+package duke;
+
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+/**
+ * The main application.
+ */
+public class Duke {
+
+    public final String FILE_NAME;
+    private final Storage FO;
+    private final TaskList TASK_LIST;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/User.jpeg"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/Duke.jpeg"));
+
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+
+    /**
+     * Returns a Duke instance.
+     *
+     * @param fileName The name of the file to store the tasks.
+     * @param fo       The FileOperator to perform file manipulation.
+     * @param taskList The TaskList instance to manage all tasks.
+     */
+    public Duke(String fileName, Storage fo, TaskList taskList) {
+        FILE_NAME = fileName;
+        this.FO = fo;
+        this.TASK_LIST = taskList;
+        startUp();
+    }
+
+    /**
+     * Used to load tasks from data.txt when Duke first starts up.
+     */
+    public void startUp() {
+        Ui ui = new Ui(FILE_NAME);
+        ui.printWelcomeMessage();
+
+        // Read from file and load into tasks
+        File file;
+
+        try {
+            file = new File(FILE_NAME);
+            if (file.createNewFile()) {
```

Comment:
Not sure if this is a coding standard violation, though if it is I can propose one (obvious) change:

~~~java

boolean fileIsValid = file.createNewFile();
if (fileIsValid) {
..

~~~

Edit: noticed several other instances using boolean return from functions  in if/while statements. Not sure if they will be that pedantic but I suppose apply above changes similarly.